pagination/README.md

```diff
@@ -0,0 +1,100 @@
+## Package `pagination` v0
+
+This is a pagination library for Go.
+
+This library provides a data structure to transport paginated data. It is useful when you need to return a page of items
+along with the total number of items and other Metadata.
+
+**What this library does:**
+
+* it provides a data structure to transport paginated data.
+* it provides functions to create "paginated data" and "page request" structures.
+* it accepts data as a slice of any type. The data supplied must be 1 page of items.
+
+**What this library does NOT do:**
+
+* it is NOT responsible for fetching the data from the database.
+* it is NOT responsible for splitting a result set into pages.
+
+The `PageRequest` struct is used to define the page number and the number of items per page. Used when requesting a page
+of items.
+
+The `Paginated` struct is used to return a page of items along with the total number of items.
+
+// Example data returned
+
+```json 
+{
+  "data": [],
+  "meta": {
+    "current_page": 1,
+    "next_page": 1,
+    "per_page": 20,
+    "prev_page": 1,
+    "total_count": 0,
+    "total_pages": 1
+  }
+}
+```
+
+### Usage
+
+#### Example of the first page of data of a small dataset:
+
+In this example, we create a new `Paginated` struct with a slice of `Item` and a `PageRequest` struct.
+
+* The total number of items in the database is 4.
+* The `PageRequest` struct specifies that we want the first page with 10 items per page.
+* The `items` slice contains 4 items (the same as the number of total items, but less than a full page).
+
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/Scalingo/go-utils/pagination"
+)
+
+type Item struct {
+	ID   int
+	Name string
+}
+
+func main() {
+	items := []Item{
+		{ID: 1, Name: "Item 1"},
+		{ID: 2, Name: "Item 2"},
+		{ID: 3, Name: "Item 3"},
+		{ID: 4, Name: "Item 4"},
+	}
+
+	p := pagination.New[[]Item](items, pagination.NewPageRequest(1, 10), 4)
```

Comment:
suggestion: the use of `len` makes it clear where does the `4` comes from.
suggestion: the use of  intermediary variables makes it explicit what they contain.

```suggestion
	page := 1
	perPage := 10
	p := pagination.New[[]Item](items, pagination.NewRequest(page, perPage), len(items))
```